### DISCUSSION TITLE
Getting: "Invalid token: token contains an invalid number of segments"

### DISCUSSION THREAD

**Discussion by Perfecto-always** [supabase/supabase#6038] [Questions] [answered]
This error occurred in the post route. 
```js
routes.post("/create", async (req, res) => {
  // this is the default token name set by supabase
  const token = req.cookies["sb:token"];
  const { user, error } = await supabase.auth.api.getUser(token);

  // If not present returns 401 with error message
  if (error) return res.status(401).json({ error });
  // Else do something else
}
```
I don't know what else is needed for debugging or reproducing the error. This is the error image 👇
![image](https://user-images.githubusercontent.com/74700079/159859859-e46287bc-a02c-4ab7-8732-3638f16fc9c1.png)
Thanks for help in advance🙏

**Comment by Perfecto-always** [comment, marked as answer]
Also this seems to only be an error in production, 'cuz it is working perfectly fine in development

**Reply by steve-chavez** [reply]
Seems the error comes from https://github.com/supabase/gotrue, maybe try reporting it there.

I think the token/cookie is malformed for these cases.

**Reply by Perfecto-always** [reply]
I think I found where the issue was, it on my part. It's something I didn't noticed, the problem was the difference in origin/domain. 

Is there any way to have `sameSite = none` and not `lax` when setting cookie on frontend?

**Reply by fanckush** [reply]
@Perfecto-always have you found a solution to this?
